test/e2e-wi-tests.sh

```diff
@@ -0,0 +1,141 @@
+  fi
+}
+
+# Tear down resources required for Control Plane setup.
+function control_plane_teardown() {
+  # When not running on Prow we need to delete the service accounts and namespaces created
+  if (( ! IS_PROW )); then
+    echo "Tear down ServiceAccount for Control Plane"
+    gcloud iam service-accounts keys delete -q ${CONTROL_PLANE_SERVICE_ACCOUNT_KEY} \
+      --iam-account=${CONTROL_PLANE_SERVICE_ACCOUNT}@${E2E_PROJECT_ID}.iam.gserviceaccount.com
+    gcloud projects remove-iam-policy-binding ${E2E_PROJECT_ID} \
+      --member=serviceAccount:${CONTROL_PLANE_SERVICE_ACCOUNT}@${E2E_PROJECT_ID}.iam.gserviceaccount.com \
+      --role roles/pubsub.admin
+    gcloud projects remove-iam-policy-binding ${E2E_PROJECT_ID} \
+      --member=serviceAccount:${CONTROL_PLANE_SERVICE_ACCOUNT}@${E2E_PROJECT_ID}.iam.gserviceaccount.com \
+      --role roles/pubsub.editor
+    gcloud projects remove-iam-policy-binding ${E2E_PROJECT_ID} \
+      --member=serviceAccount:${CONTROL_PLANE_SERVICE_ACCOUNT}@${E2E_PROJECT_ID}.iam.gserviceaccount.com \
+      --role roles/storage.admin
+    gcloud projects remove-iam-policy-binding ${E2E_PROJECT_ID} \
+      --member=serviceAccount:${CONTROL_PLANE_SERVICE_ACCOUNT}@${E2E_PROJECT_ID}.iam.gserviceaccount.com \
+      --role roles/cloudscheduler.admin
+    gcloud projects remove-iam-policy-binding ${E2E_PROJECT_ID} \
+      --member=serviceAccount:${CONTROL_PLANE_SERVICE_ACCOUNT}@${E2E_PROJECT_ID}.iam.gserviceaccount.com \
+      --role roles/logging.configWriter
+    gcloud projects remove-iam-policy-binding ${E2E_PROJECT_ID} \
+      --member=serviceAccount:${CONTROL_PLANE_SERVICE_ACCOUNT}@${E2E_PROJECT_ID}.iam.gserviceaccount.com \
+      --role roles/logging.privateLogViewer
+    gcloud projects remove-iam-policy-binding ${E2E_PROJECT_ID} \
+      --member=serviceAccount:${CONTROL_PLANE_SERVICE_ACCOUNT}@${E2E_PROJECT_ID}.iam.gserviceaccount.com \
+      --role roles/cloudscheduler.admin
+    # Remove iam.serviceAccountAdmin role to the Google service account.
+    gcloud projects remove-iam-policy-binding ${E2E_PROJECT_ID} \
+      --member=serviceAccount:${CONTROL_PLANE_SERVICE_ACCOUNT}@${E2E_PROJECT_ID}.iam.gserviceaccount.com \
+      --role roles/iam.serviceAccountAdmin
+    gcloud iam service-accounts delete -q ${CONTROL_PLANE_SERVICE_ACCOUNT}@${E2E_PROJECT_ID}.iam.gserviceaccount.com
+  fi
+}
+
+#create a cluster with Workload Identity enabled.
+initialize $@ --cluster-creation-flag "--workload-pool=\${E2E_PROJECT_ID}.svc.id.goog"
+
+# Add annotation to Kubernetes service account.
+kubectl annotate serviceaccount ${CONTROLLER_SERVICE_ACCOUNT} iam.gke.io/gcp-service-account=${CONTROL_PLANE_SERVICE_ACCOUNT}@${E2E_PROJECT_ID}.iam.gserviceaccount.com \
+--namespace ${CONTROL_PLANE_NAMESPACE}
+
+# Channel related e2e tests we have in Eventing is not running here.
+go_test_e2e -timeout=20m -parallel=12 ./test/e2e -workloadIdentity=true -pubsubServiceAccount=${PUBSUB_SERVICE_ACCOUNT} || fail_test
```

Comment:
Remove the pubsub channel tests related on eventing. Eventing test lib only accepts channel with ChannelTypeMeta, but we also need `Spec.serviceAccount` to make Workload Identity work for channel. Is there any way to also include channel's `Spec.serviceAccount` when creating broker?

Reply:
Yeah it seems we need to do some refactor to the existing e2e tests. Instead of only including `Spec.serviceAccount`, I think we can include the whole `Spec`, and set it in https://github.com/knative/eventing/blob/master/test/lib/duck/resource_creators.go#L65.